### PR TITLE
 Device Mode Support for Flutter SDK

### PR DESCRIPTION
**Pull request by desusai7**
## Description of the change

> Initial changes for Device Mode Support

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: rudder_plugin/CHANGELOG.md [hunk 1]
```diff
@@ -123,4 +123,10 @@ build()
...
 ### 2.1.1
-   Incorporated rudder web plugin 2.1.1
\ No newline at end of file
+   Incorporated rudder web plugin 2.1.1
+
+### 2.1.2-dev.1 
+    Updated all the dependent packages to dev to verify the Device Mode Implementation
+
+### 2.1.2-dev.2
+    Updated the rudder_plugin_android & rudder_plugin_ios versions
```
Comment (pallabmaiti) [inline rudder_plugin/CHANGELOG.md:132]: Add the changelog here.

**Review by pallabmaiti — CHANGES_REQUESTED** [review]
1. Can you change every summary, description, author, homepage of `podspec` files? For example: `s.summary = 'A new Flutter plugin project.'`
2. I can see lots of `gitignore`. Can you remove all & keep only one?

File: rudder_plugin/CHANGELOG.md (hunk 1 above)
Comment (desusai7) [reply rudder_plugin/CHANGELOG.md]: Added

**Comment by desusai7** [comment]
1. Updated the podspec files of all the iOS modules of the Flutter Device Mode
2. Yes, there were multiple .gitignore files, as they were automatically generated by flutter, and thought it would be better to keep them as they were, as each of them are dealing with the platform specific files, but yeah we can merge them if we want to.

**Review by pallabmaiti — APPROVED** [review]

**Pull request merged**